### PR TITLE
Renovated Editors: fix 'strict-boolean-expressions' lint rule

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: js/renovation/ui/button.tsx [hunk 1]
```diff
@@ -79,7 +79,7 @@ export const viewFunction = (viewModel: Button): JSX.Element => {
         {ButtonTemplate && (<ButtonTemplate data={{ icon, text, ...templateData }} />)}
         {!ButtonTemplate && children}
         {isIconLeft && iconComponent}
-        {renderText && (<span className="dx-button-text">{text}</span>)}
+        {typeof renderText === 'string' && renderText && (<span className="dx-button-text">{text}</span>)}
```
Comment (ksercs) [inline js/renovation/ui/button.tsx]: Let's not use `typeof`. Why don't we can just check that `renderText !== undefined`?

File: js/renovation/ui/button.tsx [hunk 2]
```diff
@@ -97,11 +97,11 @@ export const viewFunction = (viewModel: Button): JSX.Element => {
 
 @ComponentBindings()
 export class ButtonProps extends BaseWidgetProps {
-  @OneWay() activeStateEnabled?: boolean = true;
+  @OneWay() activeStateEnabled = true;
```
Comment (ksercs) [inline js/renovation/ui/button.tsx:100]: We should not remove type

File: js/renovation/ui/button.tsx (hunk 1 above)
Comment (Zedwag) [reply js/renovation/ui/button.tsx]: yes, we can

File: js/renovation/ui/button.tsx (hunk 2 above)
Comment (Zedwag) [reply js/renovation/ui/button.tsx:100]: auto removing by linter

**Review by ksercs — DISMISSED** [review]

File: js/renovation/ui/button.tsx [hunk 3]
```diff
@@ -79,7 +79,7 @@ export const viewFunction = (viewModel: Button): JSX.Element => {
         {ButtonTemplate && (<ButtonTemplate data={{ icon, text, ...templateData }} />)}
         {!ButtonTemplate && children}
         {isIconLeft && iconComponent}
-        {renderText && (<span className="dx-button-text">{text}</span>)}
+        {renderText !== false && renderText !== '' && (<span className="dx-button-text">{text}</span>)}
```
Comment (ksercs) [inline js/renovation/ui/button.tsx]: shouldn't we make `renderText` boolean?

Comment (Zedwag) [reply js/renovation/ui/button.tsx]: yeah, right

**Review by ksercs — DISMISSED** [review]

**Review by churkin — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**